source/extensions/tracers/zipkin/span_context_extractor.cc

```diff
@@ -226,6 +258,47 @@ SpanContextExtractor::extractSpanContextFromB3SingleFormat(bool is_sampled) {
   return {SpanContext(trace_id_high, trace_id, span_id, parent_id, is_sampled), true};
 }
 
+std::pair<SpanContext, bool> SpanContextExtractor::convertW3CToZipkin(
+    const Extensions::Tracers::OpenTelemetry::SpanContext& w3c_context, bool fallback_sampled) {
+  // Convert W3C 128-bit trace ID (32 hex chars) to Zipkin format
+  const std::string& trace_id_str = w3c_context.traceId();
+
+  if (trace_id_str.length() != 32) {
+    throw ExtractorException(fmt::format("Invalid W3C trace ID length: {}", trace_id_str.length()));
```

Comment:
Is this code only running when a config is ingested? If not, then throwing exceptions in the data-plane path is not allowed.
I'll leave it to @wbpcode who is a code-owner of this extension to further review.